plugin/hotswap-agent-spring-plugin/src/main/java/org/hotswap/agent/plugin/spring/core/BeanFactoryProcessor.java

```diff
@@ -0,0 +1,131 @@
+package org.hotswap.agent.plugin.spring.core;
+
+import org.hotswap.agent.logging.AgentLogger;
+import org.hotswap.agent.plugin.spring.transformers.api.IPlaceholderConfigurerSupport;
+import org.hotswap.agent.plugin.spring.utils.AnnotatedBeanDefinitionUtils;
+import org.hotswap.agent.util.ReflectionHelper;
+import org.hotswap.agent.util.spring.util.ReflectionUtils;
+import org.springframework.beans.factory.annotation.AnnotatedBeanDefinition;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.config.PlaceholderConfigurerSupport;
+import org.springframework.beans.factory.support.*;
+import org.springframework.context.annotation.ScannedGenericBeanDefinition;
+import org.springframework.context.support.GenericApplicationContext;
+import org.springframework.util.StringValueResolver;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.function.Predicate;
+
+public class BeanFactoryProcessor {
+
+    private static AgentLogger LOGGER = AgentLogger.getLogger(BeanFactoryProcessor.class);
+
+    public static void removeBeanDefinition(DefaultListableBeanFactory beanFactory, String beanName) {
+        resetEmbeddedValueResolvers(beanFactory, beanName);
+        beanFactory.removeBeanDefinition(beanName);
+    }
+
+    public static void destroySingleton(DefaultListableBeanFactory beanFactory, String beanName) {
+        // remove embeddedValueResolvers cache in PlaceholderConfigurerSupport
+        resetEmbeddedValueResolvers(beanFactory, beanName);
+        beanFactory.destroySingleton(beanName);
+    }
+
+    public static boolean isFactoryBean(DefaultListableBeanFactory beanFactory, String beanName, RootBeanDefinition beanDefinition) {
+        Method isFactoryBeanMethod = ReflectionUtils.findMethod(beanFactory.getClass(), "isFactoryBean", String.class, RootBeanDefinition.class);
+        if (isFactoryBeanMethod != null) {
+            isFactoryBeanMethod.setAccessible(true);
+            try {
+                return (boolean) isFactoryBeanMethod.invoke(beanFactory, beanName, beanDefinition);
+            } catch (IllegalAccessException e) {
+                LOGGER.warning("isFactoryBean error", e);
+            } catch (InvocationTargetException e) {
+                LOGGER.warning("isFactoryBean error", e);
+            }
+        }
+        if (beanName.startsWith("&")) {
+            return true;
+        }
+        return false;
+    }
+
+    public static boolean checkNeedReload(DefaultListableBeanFactory beanFactory, AbstractBeanDefinition currentBeanDefinition,
```

Comment:
I suggest to refactor this method into the following code:

```java
    public static boolean checkNeedReload(DefaultListableBeanFactory beanFactory, AbstractBeanDefinition beanDefinition,
                                          String beanName, Predicate<Constructor<?>[]> predicate) {
        if (hasFactoryMethodMetadata(beanDefinition)) {
            return false;
        }

        Constructor<?>[] constructors = determineConstructors(beanFactory, beanName);
        return constructors != null && constructors.length > 0 && predicate.test(constructors);
    }

    private static Constructor<?>[] determineConstructors(DefaultListableBeanFactory beanFactory, String beanName) {
        Class<?> beanClass = resolveBeanClass(beanFactory, beanName);
        if (beanClass == null) {
            return null;
        }
        
        Method method = ReflectionUtils.findMethod(beanFactory.getClass(),
                "determineConstructorsFromBeanPostProcessors", Class.class, String.class);
        if (method == null) {
            return null;
        }
        try {
            method.setAccessible(true);
            return (Constructor<?>[]) method.invoke(beanFactory, beanClass, beanName);
        } catch (Exception e) {
            LOGGER.error("determineConstructorsFromBeanPostProcessors error", e);
        }

        return null;
    }

    private static Class<?> resolveBeanClass(DefaultListableBeanFactory beanFactory, String beanName) {
        Method method = ReflectionUtils.findMethod(beanFactory.getClass(), "resolveBeanClass",
                RootBeanDefinition.class, String.class, Class[].class);
        if (method == null) {
            return null;
        }

        method.setAccessible(true);
        BeanDefinition bd = beanFactory.getMergedBeanDefinition(beanName);
        try {
            return (Class<?>) method.invoke(beanFactory, bd, beanName, new Class[]{});
        } catch (Exception e) {
            LOGGER.warning("resolveBeanClass error", e);
            return null;
        }
    }

    private static boolean hasFactoryMethodMetadata(AbstractBeanDefinition bd) {
        if (bd instanceof AnnotatedBeanDefinition) {
            AnnotatedBeanDefinition abd = (AnnotatedBeanDefinition) bd;
            return AnnotatedBeanDefinitionUtils.getFactoryMethodMetadata(abd) != null;
        }
        return false;
    }
```

'checkNeedReload' is not a good method name, pls. consider to rename.